src/include/crowbar/complex.rb

```diff
@@ -695,12 +617,195 @@ def StoreRepoURL(key, event)
     def HandleRepoURL(key, event)
       # store the value on exiting
       if event["ID"] == :next || event["EventReason"] == "ValueChanged"
+        @current_repo = UI.QueryWidget(Id("repos_table"), :Value) || ""
         StoreRepoURL(key, event)
         InitReposTable("repos_table")
       end
       nil
     end
 
+    def InitLocationURL(id)
+      UI.ChangeWidget(Id(id), :Value, @remote_server_url)
+      nil
```

Comment:
nil at the end is just "safe" for void methods, in practice, you can omit it as result of void methods are ignored